bot/exts/recruitment/talentpool/_review.py

```diff
@@ -0,0 +1,316 @@
+
+    async def _construct_review_body(self, member: Member) -> str:
+        """Formats the body of the nomination, with details of activity, infractions, and previous nominations."""
+        activity = await self._activity_review(member)
+        infractions = await self._infractions_review(member)
+        prev_nominations = await self._previous_nominations_review(member)
+
+        body = f"{activity}\n\n{infractions}"
+        if prev_nominations:
+            body += f"\n\n{prev_nominations}"
+        return body
+
+    async def _activity_review(self, member: Member) -> str:
+        """
+        Format the activity of the nominee.
+
+        Adds details on how long they've been on the server, their total message count,
+        and the channels they're the most active in.
+        """
+        log.trace(f"Fetching the metricity data for {member.id}'s review")
+        try:
+            user_activity = await self.bot.api_client.get(f"bot/users/{member.id}/metricity_review_data")
+        except ResponseCodeError as e:
+            if e.status == 404:
+                messages = "no"
+                channels = ""
+            else:
+                raise
+        else:
+            messages = user_activity["total_messages"]
+            # Making this part flexible to the amount of expected and returned channels.
+            first_channel = user_activity["top_channel_activity"][0]
+            channels = f", with {first_channel[1]} messages in {first_channel[0]}"
+
+            if len(user_activity["top_channel_activity"]) > 1:
+                channels += ", " + ", ".join(
+                    f"{count} in {channel}" for channel, count in user_activity["top_channel_activity"][1: -1]
+                )
+                last_channel = user_activity["top_channel_activity"][-1]
+                channels += f", and {last_channel[1]} in {last_channel[0]}"
+
+        time_on_server = humanize_delta(relativedelta(datetime.utcnow(), member.joined_at), max_units=2)
+        review = (
+            f"{member.name} has been on the server for **{time_on_server}**"
+            f" and has **{messages} messages**{channels}."
+        )
+
+        return review
+
+    async def _infractions_review(self, member: Member) -> str:
+        """
+        Formats the review of the nominee's infractions, if any.
+
+        The infractions are listed by type and amount, and it is stated how long ago the last one was issued.
+        """
+        log.trace(f"Fetching the infraction data for {member.id}'s review")
+        infraction_list = await self.bot.api_client.get(
+            'bot/infractions/expanded',
+            params={'user__id': str(member.id), 'ordering': '-inserted_at'}
+        )
+
+        if not infraction_list:
+            return "They have no infractions."
+
+        # Count the amount of each type of infraction.
+        infr_stats = list(Counter(infr["type"] for infr in infraction_list).items())
```

Comment:
```suggestion
        infr_stats = [Counter(infr["type"] for infr in infraction_list).items()]
```
List comps are generally better.